### ISSUE TITLE
Trailing stoploss overtaking ROI in backtesting.

### ISSUE THREAD

**Issue by aak-dev** [freqtrade/freqtrade#3233]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: ubuntu-1804-bionic-v20200218
  * Python Version: 3.7.6
  * CCXT version: ccxt==1.23.30
  * Branch: Master | Develop
  * Last Commit ID: ab71d1b9
  
## Step 3: Describe the problem:

In live mode the bot is selling exactly as configured but the backtester ignores the minimal roi and uses the trailing stoploss and sells at higher profit. 

### Steps to reproduce:
Config and strategy parameter
![image](https://user-images.githubusercontent.com/58189246/80569844-5b5e2980-8a17-11ea-8890-ee1b6ee6d011.png)

![image](https://user-images.githubusercontent.com/58189246/80569265-2d2c1a00-8a16-11ea-9682-56b55cfc428a.png)

### Observed Results:

  * What happened?

**backtest result**
![image](https://user-images.githubusercontent.com/58189246/80569354-6bc1d480-8a16-11ea-83ab-6b8760d36f25.png)

  * What did you expect to happen?
**live mode**
![image](https://user-images.githubusercontent.com/58189246/80569381-74b2a600-8a16-11ea-8d97-ed5767deb5a0.png)
### Relevant code exceptions or logs:
This is not isolated to one strategy it's happening on all and i have tested on the latest version the results are same.

**Comment by xmatthias** [comment]
Well this is intended behaviour.

please read [the documentation regarding backtesting assumptions](https://www.freqtrade.io/en/latest/backtesting/#assumptions-made-by-backtesting).

Stoplosses will ALWAYS be evaluated first (this includes trailing stoploss) to protect users capital.

Since we don't know what happens intra-candle during backtesting, freqtrade has to take certain assumptions. 
This problem is obviously bigger when using longer timeframes (it's less obvious when using 5m candles), since the movements / % changes become bigger with bigger timeframes.

So what happens is simply the following:
* stoploss logic pulls stoploss to "high - 0.75%".
* stoploss is evaluated (since low is < 0.75% it'll trigger).

Now if this is correct or wrong obviously depends if high happens before low or viceversa - but this information is unavailable during backtesting.


### To summarize, 

backtesting tries to be conservative, and tries to always evaluate stoplosses first.
The endresult will show you slightly lower results for backtesting, and higher profits in live. 
This is intentional, and reversing the above logic would cause the reverse behaviour if the intra-candle movements were different.

**Comment by aak-dev** [comment]
Basically i need to increase my roi and put my strategies on trailing stoploss to benefit the most.

But we should highlight this in docs to let people know stoploss including trailing stoploss have higher privilege in backtesting. And minmal roi wont be able to get executed properly in backtester.
As results are different and one might get confused.

**Comment by hroff-1902** [comment]
> But we should highlight this in docs 

👍 Pls review: https://github.com/freqtrade/freqtrade/pull/3235

**Comment by xmatthias** [comment]
>  And minmal roi wont be able to get executed properly in backtester.

Well this is not correct. It is executed properly - unless stoploss triggers first (as said above - being conservative and assuming the worse scenario).

Personally, i find 0.75% on trailing stop too tight - but in combination with the offset it MAY work ... although i think this can still trigger early - depending on your pairs.
